src/lsptoolshost/services/buildResultReporterService.ts

```diff
@@ -0,0 +1,30 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { Diagnostic, Uri } from 'vscode';
+import { RoslynLanguageServer } from '../roslynLanguageServer';
+import { CancellationToken } from 'vscode-jsonrpc';
+
+interface IBuildResultDiagnostics {
+    buildStarted(cancellationToken?: CancellationToken): Promise<void>;
+    reportBuildResult(
+        buildDiagnostics: Array<[Uri, Diagnostic[]]>,
+        cancellationToken?: CancellationToken
+    ): Promise<void>;
+}
+
+export class BuildResultDiagnostics implements IBuildResultDiagnostics {
+    constructor(private _languageServerPromise: Promise<RoslynLanguageServer>) {}
+
+    public async buildStarted(): Promise<void> {
+        const langServer = await this._languageServerPromise;
+        langServer._buildDiagnosticService.clearDiagnostics();
+    }
+
+    public async reportBuildResult(buildDiagnostics: Array<[Uri, Diagnostic[]]>): Promise<void> {
```

Comment:
I think I'd prefer we have an explicit type here, especially since we're using it across components

Reply:
Something like Array<DiagnosticInfo> - that way we can access the uri / diagnostics by name (instead of index into the tuple)

Reply:
Yes, I hear you on the tuple issue. What about a dictionary?? Or do you want an object containing a uri and a diagnostic list?

Reply:
Ah dictionary here is fine as well.

Reply:
Ahh that makes a lot more sense!